akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -671,6 +685,25 @@ private[akka] class ShardRegion(
       context.stop(self) // all shards have been rebalanced, complete graceful shutdown
     }
 
+  def startRegistration(): Unit = {
+    nextRegistrationDelay = initRegistrationDelay
+
+    register()
+    scheduleNextRegistration()
+  }
+
+  def scheduleNextRegistration(): Unit = {
+    timers.startSingleTimer(RegisterRetry, RegisterRetry, nextRegistrationDelay)
+    nextRegistrationDelay *= 2
+
+    if (nextRegistrationDelay >= retryInterval)
+      finishRegistration() // normal Retry msg will try to register
```

Comment:
I find it unnecessary complicated to follow this logic when it's first scheduling and then cancelling

E.g. retryInterval 1s

The RegisterRetry will be scheduled for 0.1s, 0.2s, 0.4s but for 0.8s it is cancelled because  nextRegistrationDelay is 1.6s.

Perhaps this would be better?
```
if (nextRegistrationDelay < retryInterval) {
  timers.startSingleTimer(RegisterRetry, RegisterRetry, nextRegistrationDelay)
  nextRegistrationDelay *= 2
}
```

no need for the finishRegistration() from here